### PR TITLE
ice-restart: fix functionality

### PR DESCRIPTION
**Pull request by fippo**
by listening to the transport selectedcandidatepairchange event.
This sample has been relying on the somewhat buggy but useful behavior
of the "legacy" iceconnectionstatechange firing a "connected" event even
if the previous state was already connected.

This behavior is no longer supported in the spec iceconnectionstatechange
event. Alternatively listen to the selectedcandidatepairchange event and
determine the availability of new candidates based on that

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -30,7 +30,7 @@
   ],
   "devDependencies": {
     "chai": "^4.3.6",
-    "chromedriver": "^98.0.1",
```
Comment (fippo) [inline package.json:33]: drive-by fix, otherwise chromedriver gets stuck on a version incompatible with the chrome version that gets installed

File: src/content/datachannel/filetransfer/js/test.js [hunk 2]
```diff
@@ -15,7 +15,7 @@ let driver;
 const path = '/src/content/datachannel/filetransfer/index.html';
 const url = `${process.env.BASEURL ? process.env.BASEURL : ('file://' + process.cwd())}${path}`;
 
-describe('datachannel basic', () => {
+describe('datachannel filetransfer', () => {
```
Comment (fippo) [inline src/content/datachannel/filetransfer/js/test.js:18]: drive-by too

File: src/content/peerconnection/restart-ice/js/main.js [hunk 3]
```diff
@@ -32,6 +32,8 @@ remoteVideo.addEventListener('loadedmetadata', function() {
   console.log(`Remote video videoWidth: ${this.videoWidth}px,  videoHeight: ${this.videoHeight}px`);
 });
 
+const useTransportIceEvent = window.RTCIceTransport && 'onselectedcandidatepairchange' in RTCIceTransport.prototype;
```
Comment (alvestrand) [inline src/content/peerconnection/restart-ice/js/main.js]: There are many ICE events (I think).
What about calling it "useTransportCandidatePairChange"? Verbose, but accurate.

**Review by alvestrand — APPROVED** [review]

File: src/content/peerconnection/restart-ice/js/main.js (hunk 3 above)
Comment (fippo) [reply src/content/peerconnection/restart-ice/js/main.js]: useSelectedCandidatePairChange
